### PR TITLE
fix: sidebar z-index

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Introduced a z-index for the sidebar. The sidebar is meant to be on top of everything to allow user to navigate wherever they are.
- Issue was they were at z-index 3.

### Describe what this PR does
- Short and concise, bullet points can help
- Screenshots if applicable can also help

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1579 #done

### CODE REVIEW

**Review by rebelchris — COMMENTED** [review]
I'm not 100% convinced the sidebar had a very specific z-index for mobile/other elements.
Can you just do some extended tests?

It has quite high impact if we mess up the index

**Comment by sshanzel** [comment]
> I'm not 100% convinced the sidebar had a very specific z-index for mobile/other elements. Can you just do some extended tests?
> 
> It has quite high impact if we mess up the index

The z-index is not even specific to mobile. It is just the value across sizes.

image

**Comment by rebelchris** [comment]
I know its not specific, its also not really a debate.
I expect a thorough test before we merge on different screens 🙏

**Comment by sshanzel** [comment]
> I know its not specific, its also not really a debate.
> I expect a thorough test before we merge on different screens 🙏

I mean, I agree that we will have to do a lot of testing before we merge, but just wanted to mention that the z-index existed as you mentioned you were not convinced it had a specific value.

**Comment by sshanzel** [comment]
@rebelchris have done end-to-end tests, from post-modal opening auth to the main concern with the Squad checklist tooltip.

This should also fix the long-standing devcard issue.

File: packages/shared/src/components/modals/Prompt.tsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ export function PromptElement(props: Partial<ModalProps>): ReactElement {
       size={promptSize}
       onRequestClose={onFail}
       className={className.modal}
-      overlayClassName="!z-[100]"
+      overlayClassName="!z-max"
```
Comment (rebelchris) [inline packages/shared/src/components/modals/Prompt.tsx:38]: Why do we not use modal here?

**Review by rebelchris — APPROVED** [review]
Looks alright,
Maybe if we merge to mvp intermediate let me know and we can thoroughly test 👍

One non blocking question

File: packages/shared/src/components/modals/Prompt.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/modals/Prompt.tsx:38]: Not really sure, probably as it should be way above the modal when confirming to close a modal.

**Comment by sshanzel** [comment]
> Maybe if we merge to mvp intermediate let me know and we can thoroughly test 👍

Yep, this is only pointed to Squads MVP intermediate branch.

File: packages/shared/src/components/modals/Prompt.tsx (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/components/modals/Prompt.tsx:38]: Alright will test out if i find any weird stuff once merged

**Pull request merged**